includes/modules/bootstrap/categories_tabs.php

```diff
@@ -1,42 +1,45 @@
 <?php
-/**
- * categories_tabs.php module
- *
- * @package templateSystem
- * @copyright Copyright 2003-2018 Zen Cart Development Team
- * @copyright Portions Copyright 2003 osCommerce
- * @license http://www.zen-cart.com/license/2_0.txt GNU Public License V2.0
- * @version $Id: Scott C Wilson Sat Aug 25 07:25:23 2018 -0400 Modified in v1.5.6 $
- */
-if (!defined('IS_ADMIN_FLAG')) {
-  die('Illegal Access');
-}
-$order_by = " order by c.sort_order, cd.categories_name ";
+    /**
+     * categories_tabs.php module
+     *
+     * @package   templateSystem
+     * @copyright Copyright 2003-2018 Zen Cart Development Team
+     * @copyright Portions Copyright 2003 osCommerce
+     * @license   http://www.zen-cart.com/license/2_0.txt GNU Public License V2.0
+     * @version   $Id: Scott C Wilson Sat Aug 25 07:25:23 2018 -0400 Modified in v1.5.6 $
+     */
+    if (!defined('IS_ADMIN_FLAG')) {
+        die('Illegal Access');
+    }
+    $order_by = " order by c.sort_order, cd.categories_name ";
 
-$categories_tab_query = "select c.sort_order, c.categories_id, cd.categories_name from " .
-TABLE_CATEGORIES . " c, " . TABLE_CATEGORIES_DESCRIPTION . " cd
-                          where c.categories_id=cd.categories_id and c.parent_id= '0' and cd.language_id='" . (int)$_SESSION['languages_id'] . "' and c.categories_status='1'" .
-$order_by;
-$categories_tab = $db->Execute($categories_tab_query);
+    $includeAllCategories = false;
```

Comment:
Let's add `$zca_include_zero_product_categories` (default to `false`) to `/includes/extra_datafiles/dist.site-specific-bootstrap-settings.php` and then use this here:
```php
$includeAllCategories = $zca_include_zero_product_categories ?? false;
```

Reply:
hmmm...  i screwed that up.  i wanted the default to be `true` so as to not change the default behavior.  but if you want to change the default to not include categories with 0 products (of which i am a fan), than `false` is the way to go...